proposals/NativeInt.md

```diff
@@ -148,14 +163,18 @@ In cases where there are two overloads, one for `nint` and one for `nuint`, the
 | `<<` | `native` | `int` | `native` | `shl` |
 | `>>` | `native` | `int` | `native` | `shr` / `shr.un` |
 
-For some binary operators, the IL operators support additional operand types (see [ECMA-335](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf)).
+For some binary operators, the IL operators support additional operand types
+(see [ECMA-335](https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-335.pdf) III.1.5 Operand type table).
 But the set of operand types supported by C# is limited for simplicity and for consistency with existing operators in the language.
 
 Lifted versions of the operators, where the arguments and return types are `nint?` and `nuint?`, are supported.
 
 Compound assignment operations `x op= y` where `x` or `y` are native ints follow the same rules as with other primitive types with pre-defined operators.
 Specifically the expression is bound as `x = (T)(x op y)` where `T` is the type of `x` and where `x` is only evaluated once.
 
+The shift operators should mask the number of bits to shift appropriately
+(see [shift operators](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#shift-operators) in C# spec).
```

Comment:
Tomorrow we should be reviewing https://github.com/dotnet/corefx/issues/20256, part of which is exposing a new member on `IntPtr` and `UIntPtr` that would allow this. Such a member (currently going to propose `ShiftMask`, as I've not gotten any better suggestions) would be a runtime (but not a compile time) constant.

This would allow the compiler to emit `nintValue << (shiftCount & nint.ShiftMask)`.

The alternative would be for the compiler to emit `nintValue << (shiftCount & ((sizeof(nint) * 8) - 1))` which does the same thing but doesn't put a dependency on the framework exposing a new API.